### DISCUSSION TITLE
importing sqlmodel into version file in case of using sqlmodel as ORM

### DISCUSSION THREAD

**Discussion by serge-eric-kalaga** [sqlalchemy/alembic#1404] [Usage Questions] [answered]
In case of using sqlmodel as ORM, add sqlmodel import in version file

Exemple :

```python
from typing import Sequence, Union

from alembic import op
import sqlalchemy as sa
import sqlmodel

# revision identifiers, used by Alembic.
revision: str = '002aaeee1534'
down_revision: Union[str, None] = None
branch_labels: Union[str, Sequence[str], None] = None
depends_on: Union[str, Sequence[str], None] = None

def upgrade() -> None:
    op.add_column('task', sa.Column('name', sqlmodel.sql.sqltypes.AutoString(), nullable=True))
```

**Comment by CaselIT** [comment]
Hi,

Is there a question there? Sorry but I could not find one.

If you are asking if you can import sqlmodel in the revision file I don't see why not

**Reply by serge-eric-kalaga** [reply]
I'm working on a project where I'm using SQLModel instead of SQLAlchemy.
I added Alembic in this project. But I notice that after each command I have to manually import SQLModel into the migration file (release file).
This is just an improvement suggestion to automate SQLModel import for SQLModel users who want to use Alembic.
THANK YOU.

**Reply by CaselIT** [reply]
You are free to edit the `script.py.mako` template that alembic init generates.

Alternatively there are applications hooks to add imports to the generated scripts during autogenerate

**Reply by serge-eric-kalaga** [reply]
Oh thank you.
I think that answers my suggestion.

**Reply by serge-eric-kalaga** [reply, marked as answer]
My `script.py.mako` file after update.

```
"""${message}

Revision ID: ${up_revision}
Revises: ${down_revision | comma,n}
Create Date: ${create_date}

"""
from typing import Sequence, Union

from alembic import op
import sqlalchemy as sa
import sqlmodel
${imports if imports else ""}

# revision identifiers, used by Alembic.
revision: str = ${repr(up_revision)}
down_revision: Union[str, None] = ${repr(down_revision)}
branch_labels: Union[str, Sequence[str], None] = ${repr(branch_labels)}
depends_on: Union[str, Sequence[str], None] = ${repr(depends_on)}


def upgrade() -> None:
    ${upgrades if upgrades else "pass"}


def downgrade() -> None:
    ${downgrades if downgrades else "pass"}

```

**Reply by CaselIT** [reply]
Looks all right. I've marked it as the answer

**Reply by serge-eric-kalaga** [reply]
Great. Thank you so much.
